process/block/shardblock.go

```diff
@@ -1349,6 +1369,33 @@ func (sp *shardProcessor) receivedMetaBlock(headerHandler data.HeaderHandler, me
 	go sp.txCoordinator.RequestMiniBlocks(metaBlock)
 }
 
+func (sp *shardProcessor) computeMissingPeerBlocks(metaBlock *block.MetaBlock, metaBlockHash []byte) {
```

Comment:
move all the data gathering to the validator statistics pre processor.

Reply:
fixed